### ISSUE TITLE
RichTextLabel tag [img] and support file "*.stex"

### ISSUE THREAD

**Issue by DimitriyPS** [godotengine/godot#16513]
Godot 3.0

Please add support for files of type "*.stex " in the RichTextLabel component (tag [img]).

**Comment by girng** [comment]
william can prob do this, although, if you a .stex, you have the .png already right? could use do that

**Comment by DimitriyPS** [comment]
I can use *.png, but... I don't want to offend anyone, I say in the format of personal opinion. New "model" of work resources are quite uncomfortable, opaque, forcing you to accumulate debris in a project, generates obscene file (additional garbage), overwriting the desired (changed) files, ignores disabling in settings - is one solid hemorrhoids.

There is no automation, there are no advantages, there are no positive aspects. Godot now does not support third-party resource formats (images, music) and this solution seems like an unsuccessful attempt to compensate for the lack of such support.

I am now exporting a fairly large unfinished project in which there are many resources. I export the resources, take them from "res://.import", and delete the original formats from the project, thereby avoiding the accumulation of debris in the project. That's why I care about the *.stex format support in RichTextLabel.

It is supposed that this rework is not time consuming. If so, please add her to the nearest milestone release.

**Comment by girng** [comment]
@DimitriyPS

yeah, you're actually not suppose to manage anything inside the folder yourself. if you delete the images inside the filesystem tab, the leftovers in .import get removed automatically (which is nice).

i still don't understand why godot 3 has the .import, i think it's pretty weird since it basically doubles my game's file size, which is kind of odd. godot 2.1 did it much better

**Comment by DimitriyPS** [comment]
> yeah, you're actually not suppose to manage anything inside the folder outside of the editor. 

I know that's the concept, but I have to ignore it. I control and clean up everything including the folder "res://.import". Folder " res://.import " I have absolutely empty.

> godot 2.1 did it much better imo. wish we could go back to that system too late now though.

Godot 2.1 supported third-party formats, Godot 3 not supported third-party formats. That is, the conversion of the original bitmap and audio formats into internal Godot formats Is a necessity. This is understandable and does not cause rejection. I was very saddened by the fact that the dedicated universal interface for importing resources was removed, which allowed not to put the source resources into the project. Instead, it is imposed as the only import system - a glitchy, pseudo-automatic hemorrhoids. (sorry if that sounds offensive, I'm just forced to work with her). It seems to me not only obvious, but also natural that both of these resource import systems do not contradict each other and can be used in parallel at the user's choice.

**Comment by girng** [comment]
>of the original bitmap and audio formats into internal Go 

@DimitriyPS yah, that's a good point. i think reduz and other devs kinda of overlooked/underestimated how easy the import font thing was for bitmaps. it was a godsend, compared to using BMFont (still can't even get the fonts looking right in that program). in fact, i actually use  Godot 2.1 just so i use get a nice, crisp .fnt file (it's better than BMFont). then the godot user has to rename it to .font too to open in Godot 3.0 i believe 

it was really nice to have that bitmap font support integrated. now it seems like they want us to use more "External tools" for stuff, which is yeah, nice, but for native bitmap font support, that fitted perfectly with the engine.. was like no reason to remove i feel. i still think that font creation panel should be added later in 3.1. 

especially because it works better than BMFont, a lot easier to use, and integrated in editor. i seen a couple facebook posts of people say "how to install bitmap font?!". wouldn't have had those questions!

**Comment by akien-mga** [comment]
> I am now exporting a fairly large unfinished project in which there are many resources. I export the resources, take them from "res://.import", and delete the original formats from the project, thereby avoiding the accumulation of debris in the project. That's why I care about the *.stex format support in RichTextLabel.

If that's your workflow, we can't really help you. That's complete backwards regarding how the new import system works in Godot 3. The original formats in your project folder are your source assets - the files in .import/ are the converted versions that match Godot's internal formats. You can't modify those, and if you delete them, they're generated automatically from the source files.
Deleting the source file to use the imported files manually is like producing binaries and then deleting the source code, considering you'll never need them again.

That's not how Godot 3 is designed to be used, so we won't add features to make it usable this way. It will always break at some point.

**Comment by DimitriyPS** [comment]
Yes, Yes... I figured out how it works, don't explain it to me. What I do does not contradict this - you just do not think when you read.

I only ask to support the native format Godot 3 in the [img] tag. Please resume this "Issues". In fact, this revision can not be difficult.

**Comment by akien-mga** [comment]
Well I can reopen it for others to give their opinion if you want, but we usually don't implement features without a use case, and here the use case is to basically ignore all the Godot design and open a can of worms, so I don't see much incentive for us to support it.

**Comment by DimitriyPS** [comment]
**akien-mga**, сan I ask you 2 things:

1. I do not see any advantages of the new system of export of resources (I see a lot of negative aspects). Am I missing something? What did the developers want to do when they did this?

2. Good, for some reason, the export system had to be done this way. Why was it necessary to eliminate the old one? It does not contradict the new and gives the choice to the user.

**Comment by KoBeWi** [comment]
Closing this for the same reason as before. We now have a new proposal repository where this issue fits perfectly, so if it's still important you can create it again there: https://github.com/godotengine/godot-proposals/issues
